plugins/core/camel/src/main/java/org/kantega/respiro/camel/CamelPlugin.java

```diff
@@ -47,7 +52,7 @@ public CamelPlugin(Collection<DataSourceInitializer> dataSourceInitializers, Col
     @Override
     public void deploy(Collection<RouteBuilder> routeBuilders) {
         try {
-            camelContext = new DefaultCamelContext();
+            camelContext = new DefaultCamelContext(simpleRegistry);
```

Comment:
Not really. But we can be sure that all RouteBuilders have been exported. So as long as you register your DataSource in the same plugin constructor as your RouteBuilder referring to it, you should be safe.